pruning/types/options.go

```diff
@@ -0,0 +1,134 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+)
+
+// PruningOptions defines the pruning strategy used when determining which
+// heights are removed from disk when committing state.
+type PruningOptions struct {
+	// KeepRecent defines how many recent heights to keep on disk.
+	KeepRecent uint64
+
+	// Interval defines when the pruned heights are removed from disk.
+	Interval uint64
+
+	// Strategy defines the kind of pruning strategy. See below for more information on each.
+	Strategy PruningStrategy
+}
+
+type PruningStrategy int
+
+// Pruning option string constants
+const (
+	PruningOptionDefault    = "default"
+	PruningOptionEverything = "everything"
+	PruningOptionNothing    = "nothing"
+	PruningOptionCustom     = "custom"
+)
+
+const (
+	// PruningDefault defines a pruning strategy where the last 362880 heights are
+	// kept where to-be pruned heights are pruned at every 10th height.
+	// The last 362880 heights are kept(approximately 3.5 weeks worth of state) assuming the typical
+	// block time is 6s. If these values do not match the applications' requirements, use the "custom" option.
+	PruningDefault PruningStrategy = iota
+	// PruningEverything defines a pruning strategy where all committed heights are
+	// deleted, storing only the current height and last 2 states. To-be pruned heights are
+	// pruned at every 10th height.
+	PruningEverything
+	// PruningNothing defines a pruning strategy where all heights are kept on disk.
+	// This is the only stretegy where KeepEvery=1 is allowed with state-sync snapshots disabled.
+	PruningNothing
+	// PruningCustom defines a pruning strategy where the user specifies the pruning.
+	PruningCustom
+	// PruningUndefined defines an undefined pruning strategy. It is to be returned by stores that do not support pruning.
+	PruningUndefined
+)
+
+const (
+	pruneEverythingKeepRecent = 2
+	pruneEverythingInterval   = 10
```

Comment:
What unit is this interval in? Nano seconds? seconds? I highly encourage defining it with the unit as a suffix or as a plain time.Duration value.

Reply:
Its in blocks, pruning interval within the SDK has been in blocks to date

Reply:
Gotcha, thanks @ValarDragon!

Reply:
There is a convention throughout the codebase to refer to these just as "keep-recent" and "interval", even on the client/config side. I think renaming it here would be odd.

If we want to do that, would probably make sense to do so in a separate PR through the codebase to limit the scope here

Reply:
Gotcha, thanks @p0mvn!